h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java

```diff
@@ -255,6 +254,10 @@ V getFormattedValue() {
     @API(help="The list of modeling steps to be used by the AutoML engine (they may not all get executed, depending on other constraints).",
             level = API.Level.expert)
     public StepDefinitionV99[] modeling_plan;
+    
+    @API(help="The list of preprocessing steps to run. Only 'targetencoding' is currently supported.",
```

Comment:
Do we want to change "targetencoding" to "target_encoding" for readability?

Reply:
I think both work as it is represented as `TargetEncoding` enum on the backend, then it should accept `targetencoding`, `target_encoding`, `TargetEncoding` and probably a few more.
I'll change the value in the generated doc though.

Reply:
@sebhrusen Did you want to change the Java parameter definition to use `"target_encoding"` instead of `"targetencoding"` too?  Or are you keeping it "targetencoding" because the enum name is `"TargetEncoding"` on the back-end?  (If so, should we just change to `"TargetEncoding"` here for Java users?)